numpy/random/src/distributions/distributions.c

```diff
@@ -227,7 +227,7 @@ float random_gauss_zig_f(bitgen_t *bitgen_state) {
     x = rabs * wi_float[idx];
```

Comment:
Can wi_float be negative?

Reply:
`wi_float` is an array of positive `float`s defined in https://github.com/numpy/numpy/blob/master/numpy/random/src/distributions/ziggurat_constants.h

Reply:
Why is rabs signed here?

Reply:
You would have to ask the original author.  I saw that the variable was being used elsewhere, but I didn't look too closely at how it was being used and didn't investigate changing its type.

Over in the [NumPy C Style Guide](https://github.com/numpy/numpy/blob/master/doc/C_STYLE_GUIDE.rst.txt), it says *"No compiler warnings with major compilers (gcc, VC++, a few others). Note: NumPy still produces compiler warnings that need to be addressed."*  It would be great if at least the warnings in *new* code were fixed before the code was committed.

Reply:
Given that this is used only in two places, one is bitwise, and the other is a cast - I think we should just change the type

Reply:
Done.